rust/openvasd/src/scan.rs

```diff
@@ -56,7 +58,14 @@ impl<T> From<PoisonError<T>> for Error {
 impl OSPDWrapper {
     /// Creates a new instance of OSPDWrapper
     pub fn new(socket: PathBuf) -> Self {
-        Self { socket }
+        Self {
+            socket,
+            r_timeout: None,
+        }
+    }
+
+    pub fn set_read_timeout(&mut self, secs: Duration) {
```

Comment:
why not just add it in `new`? As far as I can see the only way to change the timeout is in the configuration, environment variable and via argument all of those are just read within start period.

From that understanding I don't understand why doing it via a setter rather than on constructing.